### DISCUSSION TITLE
v3.0.0-rc.10

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#16427] [Announcements]
[**🗒️ Release Notes**](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.10)

## 🚀 How to Upgrade

> **Note**
> Make sure to recreate the lock file in the project in case of any issues after the upgrade.

- **Automated:** `npx nuxi@latest upgrade --force`
- **Manual:** Bump `nuxt` dependency to `3.0.0-rc.10` and then use `npx nuxi@latest cleanup` to cleanup any local caches

### Critical Styles are Inlined

[#6755](https://github.com/nuxt/framework/issues/6755), [#7160](https://github.com/nuxt/framework/pull/7160)

Global styles and used component styles are now automatically inlined when server-side rendering a page. This feature helps to improve the [First Contentful Paint (FCP)](https://web.dev/fcp/) metric.

🧪 You can disable the feature from `nuxt.config` using `experimental: { inlineSSRStyles: false }` in case of any issues.

### Full Static Generation with Payload Rendering

[#6411](https://github.com/nuxt/framework/issues/6411), [#6455](https://github.com/nuxt/framework/pull/6455)

In Nuxt 2, we introduced an amazing feature called [Full Static Generation](https://nuxtjs.org/announcements/going-full-static/). When using `nuxt generate`, the payload of each page containing asyncData and state is extracted to a `.js` chunk and we can deploy output to any static hosting without requiring a hosted API server. This feature also introduced performance benefits to reduce page size and allow smartly prefetching payload of next pages ahead of time when using `nuxt generate` but kept limited to it. 

This feature is now back in Nuxt 3 but much better! Payload is not only extracted during prerendering phase (`nuxt generate`) but also can be rendered on demand by simply appending `/_payload.js` to the end of any URL. This made implementation much simpler and also unlocks future development to enable payload rendering for **hybrid** static server and **incremental** generated pages. Followup [#6411](https://github.com/nuxt/framework/issues/6411) for the roadmap.

### Link Prefetching

[#4329](https://github.com/nuxt/framework/pull/4329)

Another goodie ported from Nuxt 2, is automatically prefetching the next pages when a `<NuxtLink>` is in the viewport.

This feature is integrated with vue-router to prefetch components of the next route and also payload extraction to prefetch the payload of the next pages ahead of time! You can also hook into `link:prefetch` to do more prefetches.

### Better Workspace Support

[unjs/pkg-types#34](https://github.com/unjs/pkg-types/pull/34), [#7439](https://github.com/nuxt/framework/pull/7439)

Nuxt has several configurations for directories including `rootDir` where `nuxt.config`, `package.json`, etc is, and `srcDir` which is the same as `rootDir` by default but can be customized to move project code such as `pages/` to the `src/` directory. With [Monorepo](https://monorepo.tools/) becoming more popular, it became clear we need another new option to act smarter in a monorepo. 

We have introduced a new `workspaceDir` configuration. It is **automatically detected** from rootDir using different heuristics ([how?](https://github.com/unjs/pkg-types#findworkspacedir)). This option is used to extend the search path for `node_modules` via [#7439](https://github.com/nuxt/framework/pull/7439) but we will keep spreading its use in other places.

🧪 You can manually set `workspaceDir` from `nuxt.config` in case of any issues.

### `defineNuxtConfig` is Auto Imported

[#7267](https://github.com/nuxt/framework/issues/7267), [#7485](https://github.com/nuxt/framework/pull/7485), [#7497](https://github.com/nuxt/framework/pull/7497)

Nuxt uses [unjs/jiti](https://github.com/unjs/jiti) in order to support typescript and ESM syntax for `nuxt.config`. When importing `{ defineNuxtConfig } from 'nuxt'` in Nuxt 3, it causes the whole nuxt package to be loaded. It was making startup time slower.

We have introduced a new `nuxt/config` subpath export that only exports `defineNuxtConfig` for type support but you don't even need this anymore! Just remove import and enjoy shorter syntax!

```diff
-- import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
})
```

### More Powerful `nuxt init`

We have switched to [unjs/giget](https://github.com/unjs/giget) for a much more powerful template init engine. 

Normally `nuxi init` command should work as it was before. If you were using `nuxt init org/repo` to clone the 3rd party GitHub repository, you should use `nuxi init gh:org/repo` now.

### Experimental Zero-Client-JS Mode

[#7156](https://github.com/nuxt/framework/issues/7156), [#7248](https://github.com/nuxt/framework/pull/7248)

This new experimental flag allows turning off all Nuxt client js code when server-side rendering a page. 

Using this feature is advisable for the very small minority of sites that would not benefit from client-side JS.

You can try this feature by setting `experimental: { noScripts: true }` and let us know what you think!

---

Read more about all the addition in the [🗒️ Release Notes](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.10)

**Comment by FileSoft-us** [comment]
Hi is it possible to create more complex router 
multi subdomain in nuxt config that pointing to pages

news.domain.com => news.vew 
tech.domain.com => tech.vue

**Comment by imskyyc** [comment]
Can't wait to try it out. Is there any word on a better / built-in authentication backend (JWT)?

**Comment by urbgimtam** [comment]
Looks like this "Full Static Generation" can be a real game changer. Very nice work indeed.

**Comment by riskawidarsono** [comment]
😮

**Comment by Luferov** [comment]
Really looking forward to the stable version and nuxt/apollo is almost the perfect bundle for me  ❤️

**Comment by kstraszewski** [comment]
Im getting this error when building:

```
✔ Server built in 23430ms                                                                                                                                         22:01:33
✔ Generated public .output/public                                                                                                                           nitro 22:01:33
ℹ Initializing prerenderer                                                                                                                                  nitro 22:01:33

 ERROR                                                                                                                                                      nitro 22:01:34

undefined


 ERROR  Could not load /Users/konradstraszewski/WebstormProjects/lendi-b2c/node_modules/vue/dist/vue.esm-bundler (imported by .nuxt/dist/server/server.mjs): ENOENT: no such file or directory, open '/Users/konradstraszewski/WebstormProjects/lendi-b2c/node_modules/vue/dist/vue.esm-bundler'

```

Also lots of warnings: 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/10195995/190251601-191eac1f-cf8a-495d-9555-946caeeb4eaf.png">

Development mode works great

**Reply by kstraszewski** [reply]
```
    "nuxt": "npm:nuxt3@3.0.0-rc.11-27719556.94377b1",

```
also has this problem

**Reply by kstraszewski** [reply]
ok i've had this in my file:
```
import {createApp} from 'vue/dist/vue.esm-bundler';
```
Is there any other way to import createApp?

```
import {createApp} from 'vue';
```
Does not work :(

**Reply by pi0** [reply]
Why you would need manually importing createApp?

**Comment by nlvedwin** [comment]
When doing nuxt generate this one is pointing to wrong path `import p from "/index.html/_payload.js"` instead of `/_payload.js` because of this when loading the app in a nginx server it's showing in the browser console

```Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.```

is there a way to change the **index.html part?**

<img width="1136" alt="image_ip9ri" src="https://user-images.githubusercontent.com/78135357/190283831-cd18c366-773a-4068-b9e2-783ff49c4755.png">

**Reply by danielroe** [reply]
Would you open an issue please?

**Reply by pi0** [reply]
~> https://github.com/nuxt/framework/pull/7535

**Reply by nlvedwin** [reply]
thanks guys that was fast!

**Comment by nyambe** [comment]
I am getting an error updating from rc-8


> WARN  Failed to resolve dependency: vue, present in 'optimizeDeps.include'
> 
> 
>  ERROR  The following dependencies are imported but could not be resolved:
> 
>   vue (imported by /Users/samiebuka/DEV/new/learn/nuxt-server-tron/node_modules/.pnpm/nuxt@3.0.0-rc.10/node_modules/nuxt/dist/app/entry.mjs)
> 
> Are they installed?

**Reply by pi0** [reply]
Are you using `pnpm install --shamefully-hoist` to install dependencies?

**Reply by nyambe** [reply]
I was just going to post that.  Sorry.  Yes. That fixed it!!

**Comment by marvincaspar** [comment]
First of all, thank you for your hard work and full static generation. 

I have a fully static site, but want to have one api endpoint for submitting a contact form. When I use `yarn generate` only the html is generated, but the `.output/server` folder is empty. Is there an option to also generate the api endpoint?

**Reply by semocodev** [reply]
i guess you need to run `npm run build` with ssr:false in your nuxt.config

**Reply by marvincaspar** [reply]
I have tried it but this gives me an empty index.html page.

**Reply by Mokkapps** [reply]
I have the same requirement: Full static site but I also want to deploy my server functions to Netlify. I have no idea, how this works ;-)

**Reply by danielroe** [reply]
Run `nuxt build` but set `nitro.prerender.routes`  and you will get a hybrid site.

**Reply by Mokkapps** [reply]
Thanks @danielroe! 

I already use `nitro.prerender.routes` to prerender most of my sites but I use Nuxt content and I have a lot of sub pages for my blog and Vue tips. 

As wildcards do not work inside the `routes`array, I would need to add all my 50+ pages manually....

It would be good if we could specify wildcards like: 

```
  nitro: {
    prerender: {
      routes: [
        '/',
        '/about',
        '/blog/*,
        '/vue-tips/*,
      ],
    },
  },
```

**Reply by danielroe** [reply]
You can also enable the crawler...

**Reply by Mokkapps** [reply]
Ok interesting, I haven't heard of the crawler before 🙈 

How can it be enabled?

**Reply by pi0** [reply]
Try this: 

```ts
export default defineNuxtConfig({
  nitro: {
    prerender: {
      crawlLinks: true,
      routes: ['/'],
      ignore: [
        '/news/'
      ]
    }
  }
})
```

Note: Ignore option is list of route prefixes (wildcard) to exclude from prerendering. Instead of including routes to pretender, you can exclude ones you don't want to be pre-rendered. We are working on better API via route rules to make configuration easier.

**Reply by Mokkapps** [reply]
Perfect, that works! Thanks @pi0 and @danielroe!

**Reply by marvincaspar** [reply]
Nice, works for me too 😃

**Reply by Mokkapps** [reply]
Unfortunately, the Netlify build failed after about 15 minutes: 

```
9:36:32 AM: [info] [nitro] Building Nitro Server (preset: `netlify`)
9:37:07 AM: <--- Last few GCs --->
9:37:07 AM: [1423:0x5ace900]   904971 ms: Scavenge 2007.8 (2068.0) -> 2000.3 (2068.0) MB, 4.3 / 0.0 ms  (average mu = 0.396, current mu = 0.472) allocation failure
9:37:07 AM: [1423:0x5ace900]   905016 ms: Scavenge 2008.0 (2068.0) -> 2002.1 (2068.0) MB, 5.0 / 0.0 ms  (average mu = 0.396, current mu = 0.472) allocation failure
9:37:07 AM: [1423:0x5ace900]   905066 ms: Scavenge 2008.5 (2068.0) -> 2003.5 (2084.0) MB, 10.0 / 0.0 ms  (average mu = 0.396, current mu = 0.472) allocation failure
9:37:07 AM: <--- JS stacktrace --->
9:37:07 AM: FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
9:37:07 AM:  1: 0xb02930 node::Abort() [node]
9:37:07 AM:  2: 0xa18149 node::FatalError(char const*, char const*) [node]
9:37:07 AM:  3: 0xcdd16e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
9:37:07 AM:  4: 0xcdd4e7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
9:37:07 AM:  5: 0xe94b55  [node]
9:37:07 AM:  6: 0xe95636  [node]
9:37:07 AM:  7: 0xea3b5e  [node]
9:37:07 AM:  8: 0xea45a0 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
9:37:07 AM:  9: 0xea751e v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
9:37:07 AM: 10: 0xe68792 v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [node]
9:37:07 AM: 11: 0xe630ac v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawArray(int, v8::internal::AllocationType) [node]
9:37:07 AM: 12: 0xe7849a v8::internal::Handle<v8::internal::PropertyArray> v8::internal::Factory::CopyArrayAndGrow<v8::internal::PropertyArray>(v8::internal::Handle<v8::internal::PropertyArray>, int, v8::internal::AllocationType) [node]
9:37:07 AM: 13: 0x1072a65 v8::internal::JSObject::MigrateToMap(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSObject>, v8::internal::Handle<v8::internal::Map>, int) [node]
9:37:07 AM: 14: 0x1094211 v8::internal::LookupIterator::ApplyTransitionToDataProperty(v8::internal::Handle<v8::internal::JSReceiver>) [node]
9:37:07 AM: 15: 0x10b356b v8::internal::Object::AddDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::Maybe<v8::internal::ShouldThrow>, v8::internal::StoreOrigin) [node]
9:37:07 AM: 16: 0x106a681 v8::internal::JSObject::CreateDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::Maybe<v8::internal::ShouldThrow>) [node]
9:37:07 AM: 17: 0x1073393 v8::internal::JSReceiver::SetOrCopyDataProperties(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertiesEnumerationMode, v8::base::ScopedVector<v8::internal::Handle<v8::internal::Object> > const*, bool) [node]
9:37:07 AM: 18: 0x11ee59b v8::internal::Runtime_CopyDataProperties(int, unsigned long*, v8::internal::Isolate*) [node]
```

**Reply by pi0** [reply]
Is it possible to share your repo @Mokkapps ? 🙏🏼

**Reply by Mokkapps** [reply]
Unfortunately, it's a private repo. But I can give you access, if you want to take a look at it?

**Comment by semocodev** [comment]
Great job team!!

Migrate from rc.9 to rc.10 without problems. I just run: `npx nuxi upgrade`

**Reply by pi0** [reply]
Glad hearing that! I would suggest, to also renovate the lock file from time to time. Most direct dependencies get upgraded but there are sub-dependencies with improvements and fixes that are updated only when using `upgrade --force`

**Comment by localusercamp** [comment]
**Nice work!** :tada::tada::tada:

![image](https://user-images.githubusercontent.com/45385480/190333233-5c7aff0d-43d0-476a-b523-aa757458923f.png)
*Will it be changed? =B*

**Reply by pi0** [reply]
We are still aiming for September! (Might take few weeks longer but not long!)

**Reply by localusercamp** [reply]
Niiiice

**Reply by shmidtelson** [reply]
> Niiiice

You are a quite positive person!

**Comment by KyahWill** [comment]
![image](https://user-images.githubusercontent.com/76959057/190343983-635df5d6-a64e-4b05-b6fe-a76496cc3789.png)

I'm not sure what is happening right now but for some reason, running npm run dev led me to this error

**Reply by pi0** [reply]
Have you tried with `nuxt upgrade --force` to upgrade lockfile?

**Reply by KyahWill** [reply]
Thank you for replying. I did npx nuxi-edge@latest upgrade --force and npx nuxt upgrade --force but it didn't work. I feel like there is something wrong with my package.json file because I tried making a random app and it somehow works

**Reply by danielroe** [reply]
You might try checking inside your lockfile to see what package is installing an older version of `defu`. You can also force a newer version with `package.json` overrides/resolutions if nothing else works.

**Reply by KyahWill** [reply]
I did this thank you. As it turns out, post-css 8.14 uses an older version of defu. my dev now works thank you

**Reply by flowreaction** [reply]
I am having the same issue, probably also because of postcss 8.14, did you upgrade it or how did you manage to fix this issue?

**Reply by flowreaction** [reply]
for completeness, I added this to my package.json to work around the issue:
```js
"overrides": {
        "@nuxt/postcss8": {
            "defu": "^6.0.0"
        }
}
```

**Comment by JacobSamro** [comment]
This is huge 😍

**Comment by Benbinbin** [comment]
If the rec 10 full static generation version  will lose the reactive for `useState`?

I use some states to control whole app, in rec 9 after run the `nuxt generate`  command and get the full static web pages, these states 
still work well, but in rec 10 they seem to lose the reactive when I change the route.

This is the demo repo for rec 10 and `useState` https://github.com/Benbinbin/nuxt-rec10-usestate

**Reply by Benbinbin** [reply]
a relative issue https://github.com/nuxt/framework/issues/7562

**Reply by cfab** [reply]
This is fixed in edge release...

**Comment by hrgweb** [comment]
![issue](https://user-images.githubusercontent.com/12099289/190552501-fd1d479e-0cb9-42be-92a6-0c4bd2f4a5c8.PNG)

After upgrading from rc9 to rc10 I've encounter an issue on prod, after build it doesn't work at all but it works seamlessly on dev. 

Nuxt config:
```
import { defineNuxtConfig } from 'nuxt/config'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  ssr: false,
  runtimeConfig: {
    public: {
      baseURL: process.env.API_BASE_URL,
    },
  },
})
```

Any help?

**Reply by danielroe** [reply]
This is resolved on the edge channel.

**Reply by scottmilla** [reply]
@danielroe How do I get the edge channel? My prod deploy is down as well withthe same bug as above.

**Reply by danielroe** [reply]
Check out  [these docs](https://v3.nuxtjs.org/guide/going-further/edge-channel/).

**Reply by hrgweb** [reply]
@scottmilla any luck?

**Reply by hrgweb** [reply]
@danielroe still does not work at all. I follow the **Opting Into the Edge Channel** section but no luck at all.

**Reply by scottmilla** [reply]
@danielroe @hrgweb Yes, updating to edge worked for me. Remember to delete your package-lock before reinstalling after updating to the edge in package.json

**Reply by FileSoft-us** [reply]
> Check out [these docs](https://v3.nuxtjs.org/guide/going-further/edge-channel/).
Edge channel error on Netlify generate... 

2:40:19 PM: [error] useNitro is not a function
2:40:19 PM:   at Object.invoke (node_modules/nuxi/dist/chunks/build.mjs:40:19)
2:40:19 PM:   at async Object.invoke (node_modules/nuxi/dist/chunks/generate.mjs:32:5)
2:40:19 PM:   at async _main (node_modules/nuxi/dist/cli.mjs:50:20)
2:40:19 PM: error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
2:40:19 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Reply by danielroe** [reply]
This is already resolved.

**Comment by mak33v** [comment]
RC10 empty project
add firebase & import { getAuth } from "firebase/auth"; anywhere in the application

[Vue Router warn]: uncaught error during route navigation:
[[vite-node] [ERR_INVALID_URL_SCHEME]
at ] {
  statusCode: 500,
  fatal: false,
  unhandled: false,
  statusMessage: 'Vite Error',
  __nuxt_error: true
}

**Reply by Ralph-Imholz** [reply]
Having the same issue. Have you found a solution yet?

**Reply by mak33v** [reply]
so far only

nuxt.config.js

vite: {
        devBundler: 'legacy',
},

**Reply by bbelley** [reply]
Same issue here running either rc10 or rc12. 

Did anyone ever find out why this is needed? I can get it to work using the devBunder: 'legacy' fix, but otherwise the same 500 vite error.

**Comment by QaahirStewart** [comment]
500
__vite_ssr_import_0__.defuFn is not a function

updated from rc.8 now this error any help?

**Reply by QaahirStewart** [reply]
even get the same error on clean build

**Reply by flowreaction** [reply]
I have the same error, which seems to come from the nuxt/postcss8 package which requires defu@3 which is outdated.

**Reply by QaahirStewart** [reply]
I've installed defu but same error

**Reply by flowreaction** [reply]
so my workaround is adding this to my package.json:
```js
"overrides": {
        "@nuxt/postcss8": {
            "defu": "^6.0.0"
        }
    }
```

**Comment by hussamBashat** [comment]
Any Help Please

**500**
**[vite-node] [ERR_MODULE_NOT_FOUND]** after upgrade to Nuxt 3.0.0-rc.10 [#7647](https://github.com/nuxt/framework/issues/7647#issue-1377858114)

**Reply by flowreaction** [reply]
I had the same issue, and tried the rc edge channel as described [in the nuxt docs](https://v3.nuxtjs.org/getting-started/upgrade-guide/edge-channel/#opting-into-the-edge-channel). Still had the issue, but got more error output and located the actual package which was causing it (for me, it was the motion library). After removing the library, I opted out of the edge channel again, and it worked in the regular rc10.

**Reply by hussamBashat** [reply]
Thank you it really helped me..

The `devBundler: 'legacy'` solved the **[vite-node notfound]** problem and also helped reveal the real problem with '[heroicons](https://heroicons.com/)' package.
I deleted the '_heroicons_' package and Fate is running on client without `devBundler: 'legacy'` correctly.

But also the '[headlessui](https://headlessui.com/)' package was causing problems when refreshing the page so I had to add `devBundler: 'legacy'` anyway so that the server was working properly.

**Reply by flowreaction** [reply]
glad I could help :)

**Comment by kosmeln** [comment]
The mentioned
`npx nuxi-edge@latest upgrade --force`
in the description of release notes tries to install latest edge release rc11 which is probably not what most users want. Was it meant to be just 
`npx nuxi upgrade --force` instead?

**Comment by Lucky-Strike1695** [comment]
As with every RC since rc.9, can't start nuxt in secured mode (RC.9, RC.10, edge-RC.11) : 
`nuxi dev --https --ssl-cert localhost.pem --ssl-key localhost-key.pem'`
==>
500 request to https://0.0.0.0:3000/__nuxt_vite_node__/manifest failed, reason: unable to verify the first certificate ()

**Reply by pi0** [reply]
HTTPS support for vite-node (enabled in 9 and 10) is fixed for next RC. You can try with edge-channel.

**Reply by RobertWade** [reply]
@pi0 I do have the same problem, can you elaborate a bit further on this?

**Reply by pi0** [reply]
Docs for edge channel: https://v3.nuxtjs.org/getting-started/upgrade-guide/edge-channel

You can try RC.11 fix before release.

**Comment by JFGHT** [comment]
I don't fully understand the *Better Workspace Support*.

Does this mean it will help us build composables and components that work with Nuxt apps outside the Nuxt app itself? Because that's what I'm trying to do in my monorepo but I could not find any documentation regarding this.

I only found this very useful [video tutorial](https://www.youtube.com/watch?v=Q6CK3LyYqmM) which uses RC6, Turborepo and configures Nuxt in a way it treats the components as modules plus a few other tricks.

**Comment by ashokgelal** [comment]
I upgraded from `3.0.0-rc.9` to `3.0.0-rc.10` and now nothing renders and throws the following message in the browser console:

`Uncaught TypeError: Cannot read properties of undefined (reading 'accept')` 

I tried jumping from 10 to 11 but same issue. 

This error always points to one of my vue files and nothing else. I am using a layout if that makes any difference. I understand that this is not a complete description of the issue and I apologize. I'm just wondering if someone has run into the same issue and/or if someone can point me to where I should be looking.

**Reply by SebastianBoergers** [reply]
Hey, have you found out what causes this to happen? Have had the same issue(s) and couldn't really figure it out.

**Reply by danielroe** [reply]
It is likely an outdated version of `nuxi`. It's also possible that it might be caused by you overriding `NODE_ENV` or `dev` value of your nuxt.config - but I would encourage you to open an issue with a reproduction so we can investigate.

**Reply by SebastianBoergers** [reply]
Hey, thanks a lot!

I'm rather hesitant on opening issues, since for now I assume I did something wrong, overlooked something in a changelog...and whatnot. Our project ist quite big and basically as old as Nuxt 3's initial Beta-Release. Also the upgrade from pre-RC to RC4 was also quite bumpy for us. So we may have some outdated approaches in our codebase which could cause issues.

If I can get my hands on a project-state where this issue is happening I'll try to make it reproducible with a minimal config. Can't promise that thought.

**Comment by ssoo33** [comment]
When executing 'npm run build',the terminal is warned ’Generated an empty chunk: "index" ‘，and  Refresh the environment page online,there will be 404page,what happend? can give me some help?
![image](https://user-images.githubusercontent.com/22863834/194750563-a3d3fb65-9b49-486e-9aae-5b0cdee14bd8.png)

**Reply by ssoo33** [reply]
this is my package.json
![image](https://user-images.githubusercontent.com/22863834/194750631-94e6691c-3856-4456-8508-3b261ecbfead.png)

**Reply by danielroe** [reply]
Please create an issue with a reproduction. And try using the latest RC before doing so.
